Library.java

```diff
@@ -0,0 +1,261 @@
+    }
+
+
+    public member addMember(String lastName, String firstName, String email, int phoneNo) {
+        int nextMemberId = this.getNextMemberId();
+        member newMember = new member(lastName, firstName, email, phoneNo, nextMemberId);
+        members.put(newMember.GeT_ID(), newMember);
+        return newMember;
+    }
+
+
+    public Book addBook(String a, String t, String c) {
+        int nextBookId = this.getNextBookId();
+        Book newBook = new Book(a, t, c, nextBookId);
+        catalog.put(newBook.getBookId(), newBook);
+        return newBook;
+    }
+
+
+    public member getMemberId(int memberId) {
+        if (members.containsKey(memberId))
+            return members.get(memberId);
+        return null;
+    }
+
+
+    public Book getBookId(int bookId) {
+        if (catalog.containsKey(bookId))
+            return catalog.get(bookId);
+        return null;
+    }
+
+
+    public int getLoanLimit() {
+        return LOAN_LIMIT;
+    }
+
+
+    public boolean canMemberBorrow(member currentMember) {
+        if (currentMember.Number_Of_Current_Loans() == LOAN_LIMIT)
+            return false;
+
+        if (currentMember.Fines_OwEd() >= MAX_FINES_OWED)
+            return false;
+
+        for (loan loanBeingAssessed : currentMember.GeT_LoAnS())
+            if (loanBeingAssessed.OVer_Due())
+                return false;
+
+        return true;
+    }
+
+
+    public int getNumberOfLoansAvailable(member currentMember) {
+        return LOAN_LIMIT - currentMember.Number_Of_Current_Loans();
+    }
+
+
+    public loan issueLoan(Book currentBook, member currentMember) {
+        Date dueDate = Calendar.getInstance().getDueDate(LOAN_PERIOD);
+        int nextLoanId = this.getNextLoanId();
+        loan newLoan = new loan(nextLoanId, currentBook, currentMember, dueDate);
+        currentMember.Take_Out_Loan(newLoan);
+        currentBook.borrowBook();
+        int newLoanId = newLoan.ID();
+        loans.put(newLoanId, newLoan);
+        int currentBookId = currentBook.getBookId();
+        currentLoans.put(currentBookId, newLoan);
+        return newLoan;
+    }
+
+
+    public loan getLoanByBookId(int bookId) {
+        if (currentLoans.containsKey(bookId)) {
+            return currentLoans.get(bookId);
+        }
+        return null;
+    }
+
+
+    public double calculateOverDueFine(loan currentLoan) {
+        if (currentLoan.OVer_Due()) {
+            Date loanDueDate = currentLoan.Get_Due_Date();
+            long daysOverDue = Calendar.getInstance().getDaysDifference(loanDueDate);
+            double fine = daysOverDue * FINE_PER_DAY;
+            return fine;
+        }
+        return 0.0;
+    }
+
+
+    public void dischargeLoan(loan currentLoan, boolean isDamaged) {
+        member loanee = currentLoan.Member();
+        Book loanedBook  = currentLoan.Book();
+
+        double overDueFine = calculateOverDueFine(currentLoan);
+        loanee.Add_Fine(overDueFine);

```

Comment:
All method names are to start with a lowercase letter and to be in camelBack

Reply:
Method 'Book()' belongs to loan.java, not Library.java.
Method 'Add_Fine()' belongs to member.java, not Library.java.